src/components/MovieFormInput/MovieFormInput.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { Dispatch } from "react";
+
+export enum FormActionType {
+  "UPDATE_TITLE",
+  "UPDATE_URL",
+  "UPDATE_GENRE",
+  "UPDATE_OVERVIEW",
+  "UPDATE_DATE",
+  "UPDATE_RATING",
+  "UPDATE_RUNTIME",
+  "RESET_FORM",
+  "SHOW_HINTS",
+}
+
+export interface FormAction {
+  type: FormActionType;
+  payload: string;
+}
+
+export interface MultiselectFormAction extends FormAction {
+  checked: boolean;
+}
+
+interface MovieFormBaseProps {
```

Comment:
As mentioned above, interface/type declarations just take to much space obstructing the actual logic - extra it to a separate file.